packages/react-components/src/components/Accordion/Accordion.tsx

```diff
@@ -0,0 +1,142 @@
+import * as React from 'react';
+
+import { ChevronDown } from '@livechat/design-system-icons';
+import cx from 'clsx';
+
+import { useAnimations } from '../../utils';
+import { Icon } from '../Icon';
+import { Text } from '../Typography';
+
+import { AccordionMultilineElement } from './components/AccordionMultilineElement';
+import { getLabel } from './helpers';
+import { IAccordionProps } from './types';
+
+import styles from './Accordion.module.scss';
+
+const baseClass = 'accordion';
+
+export const Accordion: React.FC<IAccordionProps> = ({
+  className,
+  children,
+  label,
+  multilineElement,
+  kind = 'default',
+  openOnInit = false,
+  isOpen,
+  onClose,
+  onOpen,
+  ...props
+}) => {
+  const isControlled = isOpen !== undefined;
+  const currentlyOpen = isControlled ? isOpen : openOnInit;
+  const contentRef = React.useRef<HTMLDivElement>(null);
+  const [size, setSize] = React.useState(0);
+  const previousSizeRef = React.useRef(size);
+  const {
+    isOpen: isExpanded,
+    isMounted,
+    setIsOpen,
+  } = useAnimations({
+    isVisible: currentlyOpen,
+    elementRef: contentRef,
+  });
+  const mergedClassName = cx(
+    styles[baseClass],
+    styles[`${baseClass}--${kind}`],
+    {
+      [styles[`${baseClass}--open`]]: isExpanded,
+    },
+    className
+  );
+
+  const handleStateChange = (state: boolean) => {
```

Comment:
I'd suggest avoiding "empty words" like "state", "value" etc, because those words alone are not providing any context to the function/variable name - because the word "state" could mean anything. So I'd suggest using in this case the word "visibility": `handleVisibilityChange`, `newVisibilityValue`.